workflows/train/transformers/glue/1-aml-finetune-job.py

```diff
@@ -0,0 +1,182 @@
+# description: Submit GLUE finetuning with Huggingface transformers library on Azure ML
+
+"""Submit GLUE finetuning with Huggingface transformers library on Azure ML.
+
+This script prepares the `src/finetune_glue.py` script to run in Azure ML.
+
+To run this script you need:
+
+    - An Azure ML Workspace
+    - A ComputeTarget to train on (we recommend a GPU-based compute cluster)
+    - Azure ML Environment:
+        - create the required python environment by running the `aml_utils.py` script
+        - This registers two environments "transformers-datasets-cpu" and "transformers-datasets-gpu"
+
+Things to try:
+
+    Different GLUE Tasks: "cola", "mnli", "mnli-mm", "mrpc", "qnli", "qqp", "rte", "sst2", "stsb", "wnli"
+    Default: "cola"
+
+    model_checkpoint: Huggingface provides pretrained models you can use, e.g.
+    - "bert-base-cased"
+    - "gpt2"
+    - "xlnet-base-cased"
+    - "roberta-base"
+    See Huggingface documenation for full set of examples: https://huggingface.co/transformers/pretrained_models.html
+    Default: "distilbert-base-uncased"
+
+Note:
+    
+    Arguments passed to `src/finetune_glue.py` will override TrainingArguments:
+    
+    https://huggingface.co/transformers/main_classes/trainer.html#trainingarguments
+
+"""
+import argparse
+from pathlib import Path
+from azureml.core import Workspace  # connect to workspace
+from azureml.core import ComputeTarget  # specify AzureML compute resources
+from azureml.core import Experiment  # connect/create experiments
+from azureml.core import Environment  # manage e.g. Python environments
+from azureml.core import ScriptRunConfig  # prepare code, an run configuration
+from azureml.core import Run  # used for type hints
+
+
+def transformers_environment(use_gpu=True):
+    """Prepares Azure ML Environment with transformers library.
+
+    Note: We install transformers library from source. See requirements file for
+    full list of dependencies.
+
+    Args:
+        use_gpu (bool): If true, Azure ML will use gpu-enabled docker image
+            as base.
+
+    Return:
+        Azure ML Environment with huggingface libraries needed to perform GLUE
+        finetuning task.
+    """
+
+    pip_requirements_path = str(Path(__file__).parent.joinpath("requirements.txt"))
+    print(f"Create Azure ML Environment from {pip_requirements_path}")
+
+    if use_gpu:
+
+        env_name = "transformers-gpu"
+        env = Environment.from_pip_requirements(
+            name=env_name, file_path=pip_requirements_path,
+        )
+        env.docker.base_image = (
+            "mcr.microsoft.com/azureml/intelmpi2018.3-cuda10.0-cudnn7-ubuntu16.04"
```

Comment:
any downside to using this base image for CPU and GPU for simplicity?

Reply:
To be honest, you wouldn't want to run these examples on anything other than a gpu. What do you think to only providing the GPU-enabled environment?